client/verta/verta/_internal_utils/_pip_requirements_utils.py

```diff
@@ -277,49 +277,72 @@ def pin_verta_and_cloudpickle(requirements):
         conflicts with the version in the current environment.
 
     """
+    # replace git-installed verta (`pip install -e verta`) with "verta"
     for i, req in enumerate(requirements):
-        if req.startswith("-e git+git@github.com:VertaAI/modeldb.git"):
-            # `pip install -e modeldb/client/verta` can make `pip freeze` return
-            # this requirement item which is unusable by `pip install`
-            # https://github.com/pypa/pip/issues/7554
-            # https://github.com/pypa/pip/issues/9625
-            # https://github.com/pypa/pip/pull/9436
-            # https://github.com/pypa/pip/pull/9822
+        if "VertaAI/modeldb.git" in req and "#egg=verta" in req:
+            # git+git is not installable
+            #     https://github.com/pypa/pip/issues/7554
+            #     https://github.com/pypa/pip/issues/9625
+            #     https://github.com/pypa/pip/pull/9436
+            #     https://github.com/pypa/pip/pull/9822
+            # git+ssh is not installable without ssh credentials
+            # TODO: maybe allow git+https because it's usable
             requirements[i] = __about__.__title__
-            break
+            continue
+
+    # inject verta
+    inject_requirement(requirements, __about__.__title__, __about__.__version__)
+
+    # inject cloudpickle
+    inject_requirement(requirements, cloudpickle.__name__, cloudpickle.__version__)
+
+
+def inject_requirement(requirements, library, version):
```

Comment:
In my own experience, functions which alter mutable arguments are harder to reason about and debug than pure functions. Unless there's a compelling performance reason, I would suggest that a method such as this should instead return the intended output instead of silently mutating inputs.

Reply:
Sounds reasonable to me!